packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -960,7 +993,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      * @deprecated - Use summarize to get summary of the container runtime.
      */
     public async snapshot(): Promise<ITree> {
-        const root: ITree = { entries: await this.dataStores.snapshot() };
+        const root: ITree = { entries: [] };
+
+        if (this.disableIsolatedChannels) {
+            root.entries = root.entries.concat(await this.dataStores.snapshot());
+        } else {
+            root.entries.push(new TreeTreeEntry(
+                channelsTreeName,
+                { entries: await this.dataStores.snapshot() },
+            ));
+
+            const metadata = this.formMetadata();
+            root.entries.push(new BlobTreeEntry(metadataBlobName, JSON.stringify(metadata)));
```

Comment:
This blob is not written when `disableIsolatedChannels` is true. `createSummary` and `summarize` write this blob irrespective of that flag.